### PR TITLE
fix(DataMapper): Fix xs:include resolution for same file name in diff…

### LINKED ISSUE #3277: DataMapper: "Ambiguous filename match" error when XSD files with same name exist in different directories
When attaching multiple XSD files to the DataMapper where two directories contain files with the same name but different content (different namespaces), the parser throws "Ambiguous filename match" even though the files are in separate directories and referenced with proper relative paths.

This appears to happen only at second-level `xs:include`.

### LINKED ISSUE #3278: DataMapper: Root element dropdown cannot show same name entries in different namespace
When there're multiple element with a same localName but in a different namespace, the root element dropdown in Attach Schema modal only show one.

### PR DESCRIPTION
**Pull request by igarashitm**
…erent path

Fixes: https://github.com/KaotoIO/kaoto/issues/3277

Also fix same localName elements in different namespace

Fixes: https://github.com/KaotoIO/kaoto/issues/3278

https://github.com/user-attachments/assets/7a5f3a3f-74ef-446d-bffe-d4ac64e62f8c

## Summary by CodeRabbit

* **New Features**
  * Improved typeahead root-element selector with clearer selection, filtering, and namespace disambiguation
  * Enhanced XML schema handling across multiple namespaces for more reliable element discovery
  * Preserves root-element selection when unrelated files are removed

* **Tests**
  * Expanded test coverage for typeahead interactions, selection behaviors, namespace edge-cases, and schema resolution scenarios

### CODE REVIEW

File: packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  const handleToggleClick = useCallback(() => {
+    if (isOpen) {
+      setIsOpen(false);
+      setFilterText('');
+    } else {
+      setFilterText('');
+      setIsOpen(true);
+      requestAnimationFrame(() => inputRef.current?.focus());
+    }
+  }, [isOpen]);
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx]: @coderabbitai no option case shows an error message and hide this dropdown, look at AttachSchemaModal.tsx line 305

File: packages/ui/src/components/Document/actions/AttachSchema/AttachSchemaModal.test.tsx [hunk 2]
```diff
@@ -497,25 +488,20 @@ describe('AttachSchemaModal', () => {
         expect(fileItem).toBeInTheDocument();
       });
 
-      const rootElementSelector = await screen.findByTestId('attach-schema-root-element-typeahead-select-input');
-      const typeaheadInput = getByLabelText(rootElementSelector, 'Attach schema / Choose Root Element');
-      expect(typeaheadInput.getAttribute('value')).toEqual('Order');
+      const typeaheadInput = await screen.findByTestId('attach-schema-root-element');
```
Comment (PVinaches) [inline packages/ui/src/components/Document/actions/AttachSchema/AttachSchemaModal.test.tsx]: Small suggestion: `typeaheadInput` and `input` are called exactly like this several times along this test file. WDYT of adding a common helper for this?

File: packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+  const toggle = useCallback(
```
Comment (mmelko) [inline packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx]: I think useCallback is not necessary here. isOpen is changing after every interaction, so useCallback is being invalidated constantly.  Plain function would work the same.

File: packages/ui/src/components/Document/actions/AttachSchema/AttachSchemaModal.test.tsx (hunk 2 above)
Comment (igarashitm) [reply packages/ui/src/components/Document/actions/AttachSchema/AttachSchemaModal.test.tsx]: Thanks, fixed :+1:

File: packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx (hunk 3 above)
Comment (igarashitm) [reply packages/ui/src/components/Document/actions/AttachSchema/TypeaheadSelect.tsx]: Thanks, fixed :+1:

**Review by mmelko — APPROVED** [review]
